### PR TITLE
fix for Missing null check after OPENSSL_zalloc #10283

### LINKED ISSUE #10283: Missing null check after OPENSSL_zalloc
Hi,

It seems that there is a potential null-dereference at line 29 if OPENSSL_zalloc returns NULL at line 26 the following code. https://github.com/openssl/openssl/blob/0a4d6c67480a4d2fce514e08d3efe571f2ee99c9/crypto/evp/keymgmt_lib.c#L26-L35

### PR DESCRIPTION
**Pull request by jayaram-intel**
fix for Missing null check after OPENSSL_zalloc 
Fixes #10283

### CODE REVIEW

**Comment by mattcaswell** [comment]
Please could you modify your commit message so that the #10283 appears on a line on its and prefixed with "Fixes #", i.e.

````
Fix for Missing null check after OPENSSL_zalloc

Fixes #10283
````

Doing it like this means that the associated issue will be closed automatically when this gets merged.
Just force push the amended commit into this PR.

**Comment by jayaram-intel** [comment]
Hi @mattcaswell ,

Just now modified commit message, Please review now.

Thanks
Jayaram Matta

File: crypto/evp/keymgmt_lib.c [hunk 1]
```diff
@@ -18,12 +18,14 @@ static OSSL_PARAM *paramdefs_to_params(const OSSL_PARAM *paramdefs)
 {
     size_t cnt;
     const OSSL_PARAM *p;
-    OSSL_PARAM *params, *q;
+    OSSL_PARAM *params= NULL, *q;
```
Comment (p-steuer) [inline crypto/evp/keymgmt_lib.c]: please put a space on both sides of =

Comment (jayaram-intel) [reply crypto/evp/keymgmt_lib.c]: Hi @p-steuer ,

Just now modified requested changes, Please review now.

Thanks
Jayaram Matta

**Comment by mattcaswell** [comment]
@p-steuer I noticed that you added the "approval: done" label - but you didn't actually approve this PR! Can you confirm the approval?

**Comment by p-steuer** [comment]
oops , i intended to approve this one. fixed.

**Comment by p-steuer** [comment]
merged, thanks.

master: https://github.com/openssl/openssl/commit/fefe3169bb12865732a3395ce43995fc661ff10b
fixed and closed: https://github.com/openssl/openssl/issues/10283

**Pull request closed without merging**